src/common/resolveEnvVariables.ts

```diff
@@ -0,0 +1,42 @@
+import fs from 'fs';
+
+type EnvObject = {
+  [variableName: string]: string;
+};
+
+//this function replaces all env variables within the docker compose yaml string with their values in the env file
+const resolveEnvVariables = (yamlText: string, filePath: string) => {
+  const envFileArray = filePath.split('/');
+  const envFilePath =
+    envFileArray.slice(0, envFileArray.length - 1).join('/') + '/.env';
+  //read envfile if there is one there
+  let envString: string;
+  try {
+    envString = fs.readFileSync(envFilePath).toString();
+  } catch (err) {
+    return yamlText;
+  }
+  let yamlTextCopy = yamlText;
+  console.log(envString);
+  //split by line
+  const envArray = envString.split('\n');
+  //remove empty last element
+  envArray.splice(-1, 1);
+  //create Object that stores the variable notation to be replace with its value
+  const envObject: EnvObject = envArray.reduce((acc: EnvObject, el: string) => {
+    const [variableName, value] = el.split('=');
+    const variableKey: string = '\\${' + variableName + '}';
```

Comment:
docker-compose.yml also supports the `$VARNAME` syntax, not just the `${VARNAME}` syntax. There's also an option for default values like so: `${VARNAME:5}`